### PR TITLE
Support `x clean --stage 1 rustc_query_impl`

### PR DESCRIPTION
**Pull request by jyn514**
Previously, clean only supported `--stage 0` for specific crates.

The new `crate_description` function generates a string that looks like
```
: {rustc_query_impl}
```

### CODE REVIEW

**Comment by jyn514** [comment]
hmm, I just realized we *also* have another way of showing individual crates:
```
Testing ["rustc_interface"] stage0 (aarch64-unknown-linux-gnu -> aarch64-unknown-linux-gnu)
```
We should probably settle on one or the other.

**Comment by jyn514** [comment]
> hmm, I just realized we _also_ have another way of showing individual crates:
> 
> ```
> Testing ["rustc_interface"] stage0 (aarch64-unknown-linux-gnu -> aarch64-unknown-linux-gnu)
> ```
> 
> We should probably settle on one or the other.

This ended up being a little bigger of a change than I expected, I moved it to a separate PR: https://github.com/rust-lang/rust/pull/106303

File: src/bootstrap/builder.rs [hunk 1]
```diff
@@ -109,6 +109,25 @@ impl RunConfig<'_> {
...
+    let mut descr = String::from(": {");
+    for krate in &*crates {
+        write!(descr, "{}, ", krate.strip_prefix("-p=").unwrap()).unwrap();
+    }
+
+    descr.pop();
+    descr.pop();
+    descr.push('}');
+    descr
+}
```
Comment (jyn514) [inline src/bootstrap/builder.rs:129]: I can remove this bit from the PR if you like; it's mostly important for https://github.com/rust-lang/rust/pull/106303 which you've already approved.

Comment (Mark-Simulacrum) [reply src/bootstrap/builder.rs:129]: No need to do that.

**Comment by Mark-Simulacrum** [comment]
r=me, modulo whether rebasing over the other PR makes sense.

**Pull request merged**